File: python/sieve-eratosthenes/sieve-eratosthenes.py
def sieve_of_eratosthenes(to_number):
    numbers = [x for x in range(2, to_number + 1)]
    output = []
    counter = 0
    while counter < len(numbers):
        if numbers[counter] != 0:
            length = len(numbers)
            jump = numbers[counter]
            for i in range(counter, length, jump):
                if i != counter:
                    numbers[i] = 0
                else:
                    output.append(numbers[i])
        counter += 1
    return output


result = sieve_of_eratosthenes(100)
print(f'Znaleziono {len(result)} liczb pierwszych. Lista liczb: {result}')

File: javascript/sieve-eratosthenes/sieve-eratosthenes.js
const sieveOfEratosthenes = (toNumber) => {
    const allNumbers = []
    const output = []
    for (let i = 2; i < toNumber; i++) {
        allNumbers.push(i)
    }
    let counter = 0

    while (counter < allNumbers.length) {
        if (allNumbers[counter]!==0) {
            let jump = allNumbers[counter]
            let length = allNumbers.length
            for (let i = counter; i < length; i += jump) {
                if (i !== counter) {
                    allNumbers[i] = 0
                } else {
                    output.push(allNumbers[i])
                }
            }
        }
        counter++
    }
    return output
}

const result = sieveOfEratosthenes(100)
console.log(`Znaleziono ${result.length} liczb pierwszych. List liczb: ${result}`)
